### PR TITLE
feat(analytics): Add reference implementation links

### PR DESCRIPTION
**Pull request by leeandher**
See: [API-2452](https://getsentry.atlassian.net/browse/API-2452)
Blocked by: https://github.com/getsentry/sentry/pull/35222

As part of the release of the reference implementation, we're adding some in app messaging to direct users to the docs and the repository directly. We're not able to track analytics greatly outside of Sentry, so we're measuring clicks here, instead of on docs.sentry.io or GitHub itself.

New UI (Updated June 9th):

image

image

image

### CODE REVIEW

**Review by mgaeta — APPROVED** [review]

**Comment by leeandher** [comment]
Adding Do Not Merge tag until we're ready for a public release

**Comment by ceorourke** [comment]
What do you think about using a banner instead of plain text in the UI? I think it's worth tagging design for their input here.

**Comment by leeandher** [comment]
Revising to add some better analytics

File: static/app/components/modals/createNewIntegrationModal.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export type CreateNewIntegrationModalOptions = {
 
 type Props = ModalRenderProps & CreateNewIntegrationModalOptions;
 
-function CreateNewIntegration({Body, Header, Footer, closeModal, orgSlug}: Props) {
+function CreateNewIntegrationModal({Body, Header, Footer, closeModal, orgSlug}: Props) {
```
Comment (leeandher) [inline static/app/components/modals/createNewIntegrationModal.tsx:17]: Just a misnomer I noticed

**Review by ceorourke — APPROVED** [review]
I'm into it

**Review by jmanhart — APPROVED** [review]
Love it!

**Pull request merged**